src/cmake_modules/FastDownwardMacros.cmake

```diff
@@ -1,57 +1,55 @@
-        "$<${msvc_cxx}:/W4;/wd4456;/wd4458;/wd4459;/wd4244;/wd4267>")
-        # Disable warnings that currently trigger in the code until we fix them.
-        #   /wd4456: declaration hides previous local declaration
-        #   /wd4458: declaration hides class member
-        #   /wd4459: declaration hides global declaration
-        #   /wd4244: conversion with possible loss of data
-        #   /wd4267: conversion from size_t to int with possible loss of data
-    target_link_libraries(fd_interface_library INTERFACE fd_warnings)
-
-    if(UNIX)
-        target_link_options(fd_interface_library INTERFACE "-g")
-    endif()
-endmacro()
-
-
 
+add_library(cxx_options INTERFACE)
+target_compile_features(cxx_options INTERFACE cxx_std_20)
+
+set(gcc_like "$<COMPILE_LANG_AND_ID:CXX,ARMClang,AppleClang,Clang,GNU,LCC>")
+set(gcc "$<COMPILE_LANG_AND_ID:CXX,GNU>")
+set(msvc "$<COMPILE_LANG_AND_ID:CXX,MSVC>")
+set(gcc_like_release "$<AND:${gcc_like},$<CONFIG:RELEASE>>")
+set(gcc_like_debug "$<AND:${gcc_like},$<CONFIG:DEBUG>>")
+set(should_use_glibcxx_debug "$<AND:${gcc_like_debug},$<BOOL:USE_GLIBCXX_DEBUG>>")
+
+target_compile_options(cxx_options INTERFACE
+    "$<${gcc_like}:-O3;-g>")
+target_compile_options(cxx_options INTERFACE
+    "$<${gcc_like_release}:-DNDEBUG;-fomit-frame-pointer>")
+target_compile_definitions(cxx_options INTERFACE
+    "$<${should_use_glibcxx_debug}:_GLIBCXX_DEBUG>")
+# Enable exceptions for MSVC.
+target_compile_options(cxx_options INTERFACE
+    "$<${msvc}:/EHsc>")
+if(UNIX) #TODO: Maybe we can also replace this with a generator expression?
+    target_link_options(cxx_options INTERFACE "-g")
+endif()
+
+add_library(cxx_warnings INTERFACE)
+target_compile_options(cxx_warnings INTERFACE
+    "$<${gcc_like}:-Wall;-Wextra;-pedantic;-Wnon-virtual-dtor>")
+
+## We ignore the warning "restrict" because of a bug in GCC 12:
+## https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651
+set(v12_or_later "$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,12>")
+set(before_v13 "$<VERSION_LESS:$<CXX_COMPILER_VERSION>,13>")
+set(bugged_gcc "$<AND:${gcc},${v12_or_later},${before_v13}>")
+target_compile_options(cxx_warnings INTERFACE
+    "$<${bugged_gcc}:-Wno-restrict>")
+
+# For MSVC, use warning level 4 (/W4) because /Wall currently detects too
+# many warnings outside of our code to be useful.
+target_compile_options(cxx_warnings INTERFACE
+    "$<${msvc}:/W4;/wd4456;/wd4458;/wd4459;/wd4244;/wd4267>")
+    # Disable warnings that currently trigger in the code until we fix them.
+    #   /wd4456: declaration hides previous local declaration
+    #   /wd4458: declaration hides class member
+    #   /wd4459: declaration hides global declaration
+    #   /wd4244: conversion with possible loss of data
+    #   /wd4267: conversion from size_t to int with possible loss of data
+target_link_libraries(cxx_options INTERFACE cxx_warnings)
+
+
+# TODO: Reopening the question whether we need it; Silvan could build in VS Code without it.
```

Comment:
I tried locally and a build with `cmake` and not specifying a `CMAKE_BUILD_TYPE` started for me, but it did not have the flags we want in a release build (`-DNDEBUG -fomit-frame-pointer`). The variable `default_build_type` that we saw on 
https://www.kitware.com/cmake-and-the-default-build-type/ is just a local variable and in the end they also change `CMAKE_BUILD_TYPE`. I think this is the way to go. We only modify the variable if it is empty, so we don't have to worry about overwriting user-set options. I'll change it back.